spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClientConfiguration.java

```diff
@@ -32,45 +33,42 @@
 @Configuration
 @ConditionalOnConsulEnabled
 @ConditionalOnProperty(value = "spring.cloud.consul.discovery.enabled", matchIfMissing = true)
-@EnableConfigurationProperties
+@EnableConfigurationProperties({LifecycleProperties.class, ConsulDiscoveryProperties.class, HeartbeatProperties.class})
 public class ConsulDiscoveryClientConfiguration {
 
 	@Autowired
 	private ConsulClient consulClient;
+	@Autowired
```

Comment:
Don't autowire beans created in this class, pass them as parameters to the method.